### ISSUE TITLE
Config flag to disable usage of endpointslices

### ISSUE THREAD

**Issue by Christoph-Raab** [metallb/metallb#936]
We're using metallb v0.10.2 in a 1.20.x cluster where endpointslices are technically available but currently disabled. 

To disable them in metallb as well we created a patch in our local metallb fork in form of a config flag in controller and speaker to make the usage configurable. 

With this issue I want to ask if that patch is of public interest and if we should contribute it to this repo. If so I will create a pull request.

**Comment by champtar** [comment]
Can you detail why the auto detection is not working ? Having a tristate flag enable/auto/disable would be nice yes.

**Comment by Christoph-Raab** [comment]
Sorry, maybe that was a bit unspecific. Endpointslices is disabled in kubeproxy via the feature gate. The endpointslices themself are not disabled and thus are created, but not used by kubeproxy and should not be used by metallb as well. 

The flag we created is a binary option where the usage of endpointslices is enabled by default and can be disabled. Our repo is private but I just forked this repo and applied the patch in my own github account, see this commit: https://github.com/Christoph-Raab/metallb/commit/396176c7da27a9f109531337d002da5de11a667e
